src/server/multi_test.cc

```diff
@@ -727,6 +727,88 @@ TEST_F(MultiTest, TestSquashing) {
   f1.Join();
 }
 
+#if 0
+TEST_F(MultiTest, MultiLeavesTxQueue) {
+  if (auto mode = absl::GetFlag(FLAGS_multi_exec_mode); mode == Transaction::NON_ATOMIC) {
+    GTEST_SKIP() << "Skipped MultiLeavesTxQueue test because multi_exec_mode is non atomic";
+    return;
+  }
+
+  // Tests the scenario, where the OOO multi-tx is scheduled into tx queue and there is another
+  // tx (mget) after it that runs and tests for atomicity.
+  absl::FlagSaver fs;
+  absl::SetFlag(&FLAGS_multi_exec_squash, false);
+
+  for (unsigned i = 0; i < 20; ++i) {
+    string key = StrCat("x", i);
+    LOG(INFO) << key << ": shard " << Shard(key, shard_set->size());
+  }
+
+  Run({"mget", "x5", "x8", "x9", "x13", "x16", "x17"});
+  ASSERT_EQ(1, GetDebugInfo().shards_count);
+
+  auto fb1 = pp_->at(1)->LaunchFiber(Launch::post, [&] {
+    // Runs multi on shard0 1000 times.
+    for (unsigned j = 0; j < 1000; ++j) {
+      Run({"multi"});
+      Run({"incrby", "x13", "1"});
+      Run({"incrby", "x16", "1"});
+      Run({"incrby", "x17", "1"});
+      Run({"exec"});
+    }
+  });
+
+  auto fb2 = pp_->at(2)->LaunchFiber(Launch::dispatch, [&] {
+    // Runs multi on shard0 1000 times.
+    for (unsigned j = 0; j < 1000; ++j) {
+      Run({"multi"});
+      Run({"incrby", "x5", "1"});
+      Run({"incrby", "x8", "1"});
+      Run({"incrby", "x9", "1"});
+      Run({"exec"});
+    }
+  });
+
+  auto check_triple = [](const RespExpr::Vec& arr, unsigned start) {
+    if (arr[start].type != arr[start + 1].type || arr[start + 1].type != arr[start + 2].type) {
```

Comment:
@romange When I run the test on my laptop it fails sometime (once 5 runs) but only on this condition, meaning the for loop of 1000 multi commands does not have any affect.

Reply:
Feel free to reduce the length it as long as it continues failing. If you can improve probability - even better.